hubblestack/files/hubblestack_nova/misc.py

```diff
@@ -278,6 +278,210 @@ def test_failure_reason(reason):
     '''
     return reason
 
+def test_mount_attrs(mount_name,attribute):
+    '''
+    Ensure that a given directory is mounted with appropriate attributes
+    '''
+    output = _execute_shell_command('mount | grep ' + mount_name)
+    if output.strip() == '':
+        return 'No match'
+    elif attribute not in output:
+        return output
+    else:
+        return True
+
+def service_not_enabled(service_name):
+    '''
+    Ensure that the given service is not running on the system. Return False if the service is running.
+    Return True otherwise
+    '''
+    output = _execute_shell_command('systemctl is-enabled ' + service_name + ' 2>/dev/null')
+    print output
+    if output.strip() == 'enabled':
+        return output
+    else:
+        return True
+
+def ensure_ssh_disabled_within_containers():
+    '''
+    Ensure that ssh deamon is not running within any container(s). 
+    '''
+    instances_raw = _execute_shell_command("docker ps --quiet")
+    instances_array = instances_raw.split("\n")
+    return_list = []
+    return_value = True
+    for instance in instances_array:
+        command = 'docker exec ' + instance + ' ps -el | grep "sshd$"'
+        output = _execute_shell_command(command)
+        if output.strip() != '':
+            return_list.append(output)
+            return_value = False
+    if return_value == False:
+        return str(return_list)
+    else:
+        return True
+
+def privileged_ports_not_mapped():
+    '''
+    Ensure that any privileged port ( below 1024 ) is not mapped within a docker container
+    '''
+    command = "docker ps --quiet | xargs docker inspect --format '{{ .Id }}: Ports={{ .NetworkSettings.Ports }}'"
+    output = _execute_shell_command(command)
+    ports = re.findall("HostPort:(\d+)]",output)
+    return_list = []
+    return_value = True
+    for port in ports:
+        if int(port) < 1024:
+            return_value  = False
+            return_list.append(port)
+
+    if return_value == False:
+        return return_list
+    else:
+        return True
+
+def ensure_sticky_bit_on_writable_directories():
+    '''
+    Ensure that sticky bit is set to all the world writable directories.
+    '''
+    command = "df --local -P | awk ' NR != 1 {printf $6}' | xargs -I '{}' find '{}' -xdev -type d \( -perm -0002 -a ! -perm -1000 \) 2>/dev/null"
+    output = _execute_shell_command(command)
+    if output.strip() == '':
+        return True
+    else:
+        return output
+
+def check_symlink(path,should_point_to):
+    '''
+    Ensure that the symlink is pointing to the path it should actually point to .
+    '''
+    command = 'find ' + path + ' -type l -ls'
+    output = _execute_shell_command( command )
+    if output.strip() == '':
+        return 'Not a symlink'
+    pointing_to = re.findall('->(.*)$',output)
+    if pointing_to[0].strip() == should_point_to.strip():
+        return True
+    else:
+        return (path + ' is incorrectly pointing to ' +  pointing_to)
```

Comment:
here 'pointing_to' stores the path to which the symlink is actually pointing. It makes more sense to print provide actual pointer location in return value ( because we already have the required pointer location in function argument in yaml file ).